ios_sample_test_app/ios_sample_test_app/ViewController.swift

```diff
@@ -0,0 +1,488 @@
+        button.titleLabel?.font = UIFont(name: "Raleway-SemiBold", size: 20)
+        button.layer.cornerRadius = 27.5
+        button.layer.cornerCurve = .continuous
+        button.addTarget(self, action: #selector(openColorBlockPage(_:)), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    private lazy var readDeepLinkButton: UIButton = {
+       let button = UIButton()
+        button.backgroundColor = UIColor(rgb: 0x0074DF)
+        button.setTitle("Read Link Data", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont(name: "Raleway-SemiBold", size: 20)
+        button.layer.cornerRadius = 27.5
+        button.layer.cornerCurve = .continuous
+        button.addTarget(self, action: #selector(openReadDeepLinkPage(_:)), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+    
+    // ---------- Navigation Functions ----------
+    
+    @objc func openColorBlockPage(_ sender: UIButton) {
+        let colorBlockPageController = ColorBlockViewController()
+        self.show(colorBlockPageController, sender: colorBlockPageButton)
+    }
+    
+    @objc func openReadDeepLinkPage(_ sender: UIButton) {
+        let readDeepLinkPageController = ReadDeepLinkViewController()
+        self.show(readDeepLinkPageController, sender: readDeepLinkButton)
+    }
+    
+    // ---------- QR Code Popup Views and Functions ----------
+    
+    // Dims screen around QR Code
+    
+    private let dimScreen: UIView = {
+        let view = UIView()
+        view.backgroundColor = .black
+        view.alpha = 0.5
+        view.isHidden = true
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    // Hides QR Code UIView on app initialization
+    
+    private let qrCode: UIView = {
+        let view = UIView()
+        view.isHidden = true
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    // Removes QR Code and dimmed background when screen is touched
+    
+    @objc(gestureRecognizer:shouldReceiveTouch:) func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldReceive touch: UITouch) -> Bool {
+            if !qrCode.isHidden {
+                qrCode.isHidden = true
+                dimScreen.isHidden = true
+            }
+            return false
+    }
+    
+    // ---------- ScrollView ----------
+    
+    lazy var scrollView: UIScrollView = {
+        let scroll = UIScrollView()
+        scroll.translatesAutoresizingMaskIntoConstraints = false
+        return scroll
+    }()
+    
+    lazy var contentView: UIView = {
+        let contentView = UIView()
+        contentView.translatesAutoresizingMaskIntoConstraints = false
+        return contentView
+    }()
+}
+
+// Enables setting colors using hex codes
+
+extension UIColor {
```

Comment:
If possible, I would just hardcode the RGB values or create a custom color asset for the hex colors the app is using instead of including this code. I recommend this site which makes it really easy www.uicolor.io.

Reply:
Hardcoded all values using that site. Thanks for that! Would have made the original coloring a lot easier